python/lsst/pipe/tasks/measurementDriver.py

```diff
@@ -0,0 +1,1052 @@
+            # Here, we iterate over references to the exposures, not copies.
+            for band, exp in self.generateProcessingUnits(mExposure, exposure):
+                self.log.info(f"Scaling variance for '{band}' band")
+                varScale = self.scaleVariance.run(exp.maskedImage)
+                exp.getMetadata().add("VARIANCE_SCALE", varScale)
+
+        # Detect sources in the image and populate the catalog.
+        if self.config.doDetect:
+            if catalog is not None:
+                raise RuntimeError(
+                    "An input catalog was given to bypass detection, but 'doDetect' is still on"
+                )
+            self.log.info(f"Running detection on a {exposure.width}x{exposure.height} pixel exposure")
+            catalog, backgroundList = self._detectSources(exposure, idGenerator)
+        else:
+            if catalog is None:
+                raise RuntimeError("Cannot run without detection if no 'catalog' is provided")
+            else:
+                self.log.info("Using detections from provided catalog; skipping detection")
+            backgroundList = None
+
+        # Deblend detected sources and update the catalog.
+        if self.config.doDeblend:
+            self.log.info(f"Deblending using '{self._Deblender}' on {len(catalog)} detection footprints")
+            if mExposure:
+                catalogs, modelData = self._deblendSources(mExposure, catalog, refBand=refBand)
+                # The deblender may not produce contiguous catalogs; ensure
+                # contiguity for the subsequent subtasks.
+                catalogs = self._makeContiguous(catalogs)
+            else:
+                self._deblendSources(exposure, catalog)
+                catalog = self._makeContiguous(catalog)
+        else:
+            self.log.info("Deblending is disabled; skipping deblending")
+
+        # Make variables we need accessible to the rest of the method.
+        if mExposure:
+            catalog = None
+        else:
+            catalogs = None
+
+        # Process catalog(s) through measurement, aperture correction, and
+        # catalog calculation subtasks.
+        for band, exp, cat in self.generateProcessingUnits(mExposure, exposure, catalogs, catalog):
+            # Set the PSF cache capacity to cache repeated PSF evaluations at
+            # the same point coming from different measurement plugins.
+            if self.config.psfCache > 0:
+                # Set a hard limit on the number of PSFs to cache.
+                exp.psf.setCacheCapacity(self.config.psfCache)
+            else:
+                # Auto-size the cache based on the number of measurement
+                # plugins. We assume each algorithm tries to evaluate the PSF
+                # twice, which is more than enough since many don't evaluate it
+                # at all, and there's no *good* reason for any algorithm to
+                # evaluate it more than once.
```

Comment:
Borrowed this from [here](https://github.com/lsst/drp_tasks/blob/15d4fdd68ac8aaa73b705fe3093f7fe28bdc657e/python/lsst/drp/tasks/forcedPhotCoadd.py#L324).